### ISSUE TITLE
QT: windeployqt fails due to wrong os type being assumed at qttools compile

### ISSUE THREAD

**Issue by ATiltedTree** [Martchus/PKGBUILDs#115]
I was trying to use `windeployqt` to package my program dynamically. While doing this I ran into two problems:
1. `/usr/x86_64-w64-mingw32/lib/qt/bin` has to be prepend to the current `PATH` for `windeployqt` to find the right qmake because otherwise it wants an ELF binary. See: [here](https://github.com/qt/qttools/blob/5051c1e601d60707ae821c0d6836e422249c3721/src/shared/winutils/utils.cpp#L502)
2. `windeployqt` also fails after fixing the first problem because it's compiles using the host compiler who does not set `WIN32` which disables `Q_OS_WIN` an therefore does not implement PE executable parsing. See: [here](https://github.com/qt/qttools/blob/5051c1e601d60707ae821c0d6836e422249c3721/src/shared/winutils/utils.cpp#L678) and [here](https://github.com/qt/qtbase/blob/13e0a36626bd75e631dd9536e795a494432b1945/src/corelib/global/qsystemdetection.h#L117)

`windeployqt` error log:
```
Unable to find dependent libraries of /mnt/Projects/APASSTools/build/bin/APASSTools.exe :Not implemented.
```
The `Not implemented` originates from [here](https://github.com/qt/qttools/blob/5051c1e601d60707ae821c0d6836e422249c3721/src/shared/winutils/utils.cpp#L976).
Probably as a fallback if you are compiling `windeployqt` on a non windows platform. But still try to use it with the 'win32-g++' mkspec.

**Comment by Martchus** [comment]
Hi, I know it is broken. I'm not using the tool myself but other users complained about it, too.

As you have already pointed out yourself, this is a limitation of Qt itself. Upstream does not care about it as they don't support cross compilation under Linux for Windows. Last time I submitted a patch so windeployqt would at least compile again, I [had to defend my use-case](https://codereview.qt-project.org/c/qt/qttools/+/251910).

Having that said, it is likely not a big a deal to fix windeployqt so it works under Linux. I'm currently not motivated to provide a patch my self but would accept contributions of course.

Then there's also [mxedeployqt](https://github.com/saidinesh5/mxedeployqt). Not sure whether it works out-of-the-box with my packaging, though.

**Comment by ATiltedTree** [comment]
I think that it currently does not make sense to fiddle around with it because [Qt 6 will fully transition to CMake](https://bugreports.qt.io/browse/QTBUG-73351). Which will also [support MinGW natively](https://bugreports.qt.io/browse/QTBUG-74612).

**Comment by Martchus** [comment]
I know and I hope the CMake transition will be an improvement. It certainly can as CMake's cross compilation support is generally good. However, I doubt that this will solve this problem magically. windeployqt will still be the same tool, just built with a different build system. I'm also wondering why you think support for MinGW will improve. Note that they in fact already support MinGW. It is just the cross compilation which is generally not supported/tested.

Regarding Qt 6 in general: I plan to create mingw-w64- packages for Qt 6 as well. However, I expect that Qt 5 will nevertheless still be around for a while so it makes generally sense to still care about it.

But maybe it makes actually most sense to forget about Qt's deployment tool entirely and come up with an independent solution at some point which will work for any Qt application but also any other non-Qt application. I've recently wrote an ELF and PE parser for tooling to help me maintaining my Arch repository. With that one could actually easily create a generic deployment tool. So maybe I'll come up with something more generic at some point.

**Comment by Martchus** [comment]
By the way, I've seen that you recently forked mariadb-connector-c. I assume you had no luck or it was not worth the effort after all?

**Comment by ATiltedTree** [comment]
I was able to get x86_64 to build but the failed at the i686 build and was unable to find a solution but I could give it another try. I think it was because of the sjlj exception handling which produced linker errors that I could not fix.

**Comment by Martchus** [comment]
Interesting. *That* problem might actually solve itself when switching to DW2 exception handling for i686. I suppose xantares will make the switch when updating to GCC 10.

**Comment by ATiltedTree** [comment]
I just did some further testing for mariadb-connector-c.
The connector itself compiles fine but qt then fails to link against it:
```
> i686-w64-mingw32-g++ -Wl,-O1,--sort-common,--as-needed -fstack-protector -Wl,-subsystem,console -mthreads -o mysql.exe main.o  -lmariadb   
> /usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: main.o:main.cpp:(.text.startup+0xc): undefined reference to `mysql_get_client_version@0'
> collect2: error: ld returned 1 exit status
> make: *** [Makefile:69: mysql.exe] Error 1
```

This is probably because of these warnings when building the connector on i686:
```
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_cancel by linking to _mariadb_cancel@4
Use --enable-stdcall-fixup to disable these warnings
Use --disable-stdcall-fixup to disable these fixups
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_connection by linking to _mariadb_connection@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_convert_string by linking to _mariadb_convert_string@28
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_free_rpl_event by linking to _mariadb_free_rpl_event@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_get_charset_by_name by linking to _mariadb_get_charset_by_name@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_get_charset_by_nr by linking to _mariadb_get_charset_by_nr@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_get_info by linking to _mariadb_get_info@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_reconnect by linking to _mariadb_reconnect@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_rpl_close by linking to _mariadb_rpl_close@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_rpl_fetch by linking to _mariadb_rpl_fetch@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_rpl_open by linking to _mariadb_rpl_open@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_stmt_execute_direct by linking to _mariadb_stmt_execute_direct@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mariadb_stmt_fetch_fields by linking to _mariadb_stmt_fetch_fields@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_affected_rows by linking to _mysql_affected_rows@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_autocommit by linking to _mysql_autocommit@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_autocommit_cont by linking to _mysql_autocommit_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_autocommit_start by linking to _mysql_autocommit_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_change_user by linking to _mysql_change_user@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_change_user_cont by linking to _mysql_change_user_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_change_user_start by linking to _mysql_change_user_start@20
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_character_set_name by linking to _mysql_character_set_name@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_client_find_plugin by linking to _mysql_client_find_plugin@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_client_register_plugin by linking to _mysql_client_register_plugin@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_close by linking to _mysql_close@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_close_cont by linking to _mysql_close_cont@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_close_start by linking to _mysql_close_start@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_commit by linking to _mysql_commit@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_commit_cont by linking to _mysql_commit_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_commit_start by linking to _mysql_commit_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_data_seek by linking to _mysql_data_seek@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_debug by linking to _mysql_debug@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_dump_debug_info by linking to _mysql_dump_debug_info@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_dump_debug_info_cont by linking to _mysql_dump_debug_info_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_dump_debug_info_start by linking to _mysql_dump_debug_info_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_embedded by linking to _mysql_embedded@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_eof by linking to _mysql_eof@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_errno by linking to _mysql_errno@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_error by linking to _mysql_error@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_escape_string by linking to _mysql_escape_string@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_field by linking to _mysql_fetch_field@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_field_direct by linking to _mysql_fetch_field_direct@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_fields by linking to _mysql_fetch_fields@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_lengths by linking to _mysql_fetch_lengths@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_row by linking to _mysql_fetch_row@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_row_cont by linking to _mysql_fetch_row_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_fetch_row_start by linking to _mysql_fetch_row_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_field_count by linking to _mysql_field_count@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_field_seek by linking to _mysql_field_seek@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_field_tell by linking to _mysql_field_tell@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_free_result by linking to _mysql_free_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_free_result_cont by linking to _mysql_free_result_cont@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_free_result_start by linking to _mysql_free_result_start@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_character_set_info by linking to _mysql_get_character_set_info@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_charset_by_name by linking to _mysql_get_charset_by_name@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_charset_by_nr by linking to _mysql_get_charset_by_nr@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_client_info by linking to _mysql_get_client_info@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_client_version by linking to _mysql_get_client_version@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_host_info by linking to _mysql_get_host_info@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_option by linking to _mysql_get_option@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_parameters by linking to _mysql_get_parameters@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_proto_info by linking to _mysql_get_proto_info@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_server_info by linking to _mysql_get_server_info@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_server_name by linking to _mysql_get_server_name@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_server_version by linking to _mysql_get_server_version@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_socket by linking to _mysql_get_socket@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_ssl_cipher by linking to _mysql_get_ssl_cipher@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_timeout_value by linking to _mysql_get_timeout_value@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_get_timeout_value_ms by linking to _mysql_get_timeout_value_ms@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_hex_string by linking to _mysql_hex_string@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_info by linking to _mysql_info@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_init by linking to _mysql_init@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_insert_id by linking to _mysql_insert_id@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_kill by linking to _mysql_kill@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_kill_cont by linking to _mysql_kill_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_kill_start by linking to _mysql_kill_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_list_dbs by linking to _mysql_list_dbs@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_list_fields by linking to _mysql_list_fields@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_list_fields_cont by linking to _mysql_list_fields_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_list_fields_start by linking to _mysql_list_fields_start@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_list_processes by linking to _mysql_list_processes@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_list_tables by linking to _mysql_list_tables@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_load_plugin_v by linking to _mysql_load_plugin_v@20
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_more_results by linking to _mysql_more_results@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_net_field_length by linking to _mysql_net_field_length@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_net_read_packet by linking to _mysql_net_read_packet@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_next_result by linking to _mysql_next_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_next_result_cont by linking to _mysql_next_result_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_next_result_start by linking to _mysql_next_result_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_num_fields by linking to _mysql_num_fields@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_num_rows by linking to _mysql_num_rows@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_options by linking to _mysql_options@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_options4 by linking to _mysql_options4@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_ping by linking to _mysql_ping@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_ping_cont by linking to _mysql_ping_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_ping_start by linking to _mysql_ping_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_query by linking to _mysql_query@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_query_cont by linking to _mysql_query_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_query_start by linking to _mysql_query_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_read_query_result by linking to _mysql_read_query_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_read_query_result_cont by linking to _mysql_read_query_result_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_read_query_result_start by linking to _mysql_read_query_result_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_connect by linking to _mysql_real_connect@32
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_connect_cont by linking to _mysql_real_connect_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_connect_start by linking to _mysql_real_connect_start@36
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_escape_string by linking to _mysql_real_escape_string@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_query by linking to _mysql_real_query@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_query_cont by linking to _mysql_real_query_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_real_query_start by linking to _mysql_real_query_start@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_refresh by linking to _mysql_refresh@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_refresh_cont by linking to _mysql_refresh_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_refresh_start by linking to _mysql_refresh_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_reset_connection by linking to _mysql_reset_connection@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_reset_connection_cont by linking to _mysql_reset_connection_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_reset_connection_start by linking to _mysql_reset_connection_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_rollback by linking to _mysql_rollback@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_rollback_cont by linking to _mysql_rollback_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_rollback_start by linking to _mysql_rollback_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_row_seek by linking to _mysql_row_seek@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_row_tell by linking to _mysql_row_tell@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_select_db by linking to _mysql_select_db@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_select_db_cont by linking to _mysql_select_db_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_select_db_start by linking to _mysql_select_db_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_send_query by linking to _mysql_send_query@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_send_query_cont by linking to _mysql_send_query_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_send_query_start by linking to _mysql_send_query_start@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_server_end by linking to _mysql_server_end@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_server_init by linking to _mysql_server_init@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_session_track_get_first by linking to _mysql_session_track_get_first@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_session_track_get_next by linking to _mysql_session_track_get_next@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_character_set by linking to _mysql_set_character_set@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_character_set_cont by linking to _mysql_set_character_set_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_character_set_start by linking to _mysql_set_character_set_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_local_infile_handler by linking to _mysql_set_local_infile_handler@24
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_server_option by linking to _mysql_set_server_option@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_server_option_cont by linking to _mysql_set_server_option_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_set_server_option_start by linking to _mysql_set_server_option_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_shutdown by linking to _mysql_shutdown@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_shutdown_cont by linking to _mysql_shutdown_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_shutdown_start by linking to _mysql_shutdown_start@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_sqlstate by linking to _mysql_sqlstate@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_ssl_set by linking to _mysql_ssl_set@24
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stat by linking to _mysql_stat@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stat_cont by linking to _mysql_stat_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stat_start by linking to _mysql_stat_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_affected_rows by linking to _mysql_stmt_affected_rows@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_attr_get by linking to _mysql_stmt_attr_get@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_attr_set by linking to _mysql_stmt_attr_set@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_bind_param by linking to _mysql_stmt_bind_param@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_bind_result by linking to _mysql_stmt_bind_result@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_close by linking to _mysql_stmt_close@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_close_cont by linking to _mysql_stmt_close_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_close_start by linking to _mysql_stmt_close_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_data_seek by linking to _mysql_stmt_data_seek@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_errno by linking to _mysql_stmt_errno@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_error by linking to _mysql_stmt_error@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_execute by linking to _mysql_stmt_execute@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_execute_cont by linking to _mysql_stmt_execute_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_execute_start by linking to _mysql_stmt_execute_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_fetch by linking to _mysql_stmt_fetch@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_fetch_column by linking to _mysql_stmt_fetch_column@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_fetch_cont by linking to _mysql_stmt_fetch_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_fetch_start by linking to _mysql_stmt_fetch_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_field_count by linking to _mysql_stmt_field_count@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_free_result by linking to _mysql_stmt_free_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_free_result_cont by linking to _mysql_stmt_free_result_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_free_result_start by linking to _mysql_stmt_free_result_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_init by linking to _mysql_stmt_init@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_insert_id by linking to _mysql_stmt_insert_id@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_more_results by linking to _mysql_stmt_more_results@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_next_result by linking to _mysql_stmt_next_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_next_result_cont by linking to _mysql_stmt_next_result_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_next_result_start by linking to _mysql_stmt_next_result_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_num_rows by linking to _mysql_stmt_num_rows@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_param_count by linking to _mysql_stmt_param_count@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_param_metadata by linking to _mysql_stmt_param_metadata@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_prepare by linking to _mysql_stmt_prepare@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_prepare_cont by linking to _mysql_stmt_prepare_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_prepare_start by linking to _mysql_stmt_prepare_start@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_reset by linking to _mysql_stmt_reset@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_reset_cont by linking to _mysql_stmt_reset_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_reset_start by linking to _mysql_stmt_reset_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_result_metadata by linking to _mysql_stmt_result_metadata@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_row_seek by linking to _mysql_stmt_row_seek@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_row_tell by linking to _mysql_stmt_row_tell@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_send_long_data by linking to _mysql_stmt_send_long_data@16
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_send_long_data_cont by linking to _mysql_stmt_send_long_data_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_send_long_data_start by linking to _mysql_stmt_send_long_data_start@20
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_sqlstate by linking to _mysql_stmt_sqlstate@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_store_result by linking to _mysql_stmt_store_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_store_result_cont by linking to _mysql_stmt_store_result_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_store_result_start by linking to _mysql_stmt_store_result_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_stmt_warning_count by linking to _mysql_stmt_warning_count@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_store_result by linking to _mysql_store_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_store_result_cont by linking to _mysql_store_result_cont@12
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_store_result_start by linking to _mysql_store_result_start@8
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_thread_end by linking to _mysql_thread_end@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_thread_id by linking to _mysql_thread_id@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_thread_init by linking to _mysql_thread_init@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_thread_safe by linking to _mysql_thread_safe@0
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_use_result by linking to _mysql_use_result@4
/usr/lib/gcc/i686-w64-mingw32/9.3.0/../../../../i686-w64-mingw32/bin/ld: warning: resolving _mysql_warning_count by linking to _mysql_warning_count@4
```
I don't know why these warnings show up on an i686 build because they don't on an x86_64 one.

**Comment by Martchus** [comment]
Looks like the 32-bit version is expected to use stdcall decorated symbols but doesn't. Maybe there's a macro supposed to expand to `__stdcall` when MariaDB is built but some `#ifdef` condition prevents it from being activated. Or they use def files for exporting symbols and the wrong def file is being used for the 32-bit build.

Note that the 64-bit build should never use stdcall decorated symbols anyways. That is the reason why it is unaffected by that problem.

See https://docs.microsoft.com/en-us/cpp/cpp/stdcall and https://docs.microsoft.com/en-us/cpp/build/reference/decorated-names

**Comment by Martchus** [comment]
By the way, thanks for looking into it again. Even if you can't resolve these issues your changes are likely already helpful. So keep your forks/branches present in any case :-)

**Comment by ATiltedTree** [comment]
The error seems to be that the def file generated by cmake is actually hand-crafted and does not contain the @n STDCALL appendix.

**Comment by Martchus** [comment]
You're right.

The export/def-file-generation is implemented here and it doesn't seem to include the STDCALL syntax: https://github.com/mariadb-corporation/mariadb-connector-c/blob/3ffa60dbea57c69926f3cfce286e9f650e3c0047/cmake/export.cmake

On the other hand, the functions are declared using the following macro:

```
#if !defined(_WIN32)
#define STDCALL
#else
#define STDCALL __stdcall
#endif
```

I wonder what happens if one would reduce that to:

```
#define STDCALL
```

It might work then. None of the symbols would be stdcall decorated anymore and the def file would match the actually present symbols.

**Comment by ATiltedTree** [comment]
That now works perfectly!
The connector and Qt build without a problem!
I can open a pull request for this repo but I can't for the other one since you would have to create that branch (v3.1.7-mingw-w64) so that I can merge into it.

**Comment by Martchus** [comment]
I've created the branch v3.1.7-mingw-w64 based on v3.1.7. You could create a PR. You could also create a PR here. I usually use https://github.com/Martchus/PKGBUILDs/blob/master/devel/import-patches.sh to export patches into this repo.

**Comment by ATiltedTree** [comment]
I just have to create a pull request since I already have them updated. PR: #116

**Comment by yan12125** [comment]
> maybe mxedeployqt just works with mingw-w64 packages as well.

Great hint! I gave it a try, and it works like a charm :)

Here is a command I used with [barrier](https://github.com/debauchee/barrier), which is patched for cross-compiling with MinGW packages in the [ownstuff] repo
```
mxedeployqt --mxepath /usr/x86_64-w64-mingw32 --mxetarget x86_64-w64-mingw32 build/bin
```
`build/bin` contains *.exe files.

**Comment by Martchus** [comment]
Nice that it works. It shouldn't make a difference for the tool whether one uses my packages or mxe. (I haven't used it myself for quite a while because I'm currently using the 2nd option which is the static version.)

Btw, I would have already updated Qt 5 to include more recent patches from KDE's fork but I'll have to fix some build errors first (likely just a missing header include).

**Comment by yan12125** [comment]
> mxedeployqt --mxepath /usr/x86_64-w64-mingw32 --mxetarget x86_64-w64-mingw32 build/bin

I run a similar command again today and got the following error:
```
[ERROR]  Error: qmake not found
```
Looks like the issue has been fixed in https://github.com/saidinesh5/mxedeployqt/commit/6224a172d06f3519fda926963b66282f5f72070c, while ownstuff/mxedeploy-git appears to point to an older commit https://github.com/saidinesh5/mxedeployqt/commit/f788bdfa3035139eef5be6d715a0f2ea75a2b320:
```
mxedeployqt-git 0.1.1.gf788bdf-1
```

**Comment by Martchus** [comment]
I've just been updating the package within the binary repository.

**Comment by yan12125** [comment]
Thanks! The updated package works as usual.

**Comment by melvyn2** [comment]
This problem affects me because I'm trying to build a rust app using Qt 5.15, and so I can't use `mxedeployqt` as it can't be used with cargo/the `cc` crate (afaik). I'm fine with having to set `PATH` myself as long as the tool does its job. Right now, I'm having to keep a second installation of Qt in my wineprefix, and I use the windows version of `windeployqt`, but this has some rough edges like not copying the DLLs for libstc++/winpthreads/libgcc.
It looks like it's only blocked by `Q_OS_WIN` not being defined, I'll see if I can make a patch to work around this.

**Comment by Martchus** [comment]
Ok, if you have any patches for me to pick up, just tell me. I suppose I should at some point pull in new patches from KDE anyways (and rebase against 5.15.5). However, I'm wondering what additional difficulty Rust brings so `mxedeployqt` cannot handle it.

**Comment by melvyn2** [comment]
Removing the `option(host_build)` builds it with mingw and makes a runnable exe, but this expects qmake to also be available as mingw exe: `Unable to query qmake: CreateProcessW failed: File not found.`
As for my original problem, I found https://forum.qt.io/topic/88525/windeployqt-exe-does-not-copy-all-needed-files ; I'll try to maybe get mingw working in wine and keep using that since it seems that windeployqt is too tied to Win32 apis to port to native and can't be used in mingw. Was it ever actually working as a native linux binary.

**Comment by melvyn2** [comment]
Ok my current solution is installing Qt and mingw-w64 in wine and making distributables with `WINEPATH='C:\mingw64\bin\' wine 'C:\Qt\5.15.2\mingw81_64\bin\windeployqt.exe' dist/example.exe --qmldir src/example/`

**Comment by Martchus** [comment]
Not nice but if it works. I suppose another approach would be adjusting `mxedeployqt` to your needs. It is just a Python script. I'm actually wondering what special requirement Rust DLLs bring.

**Comment by melvyn2** [comment]
Fiddling with `PATH` and `.cargo/config.toml`, I was able to get cargo to build using mxe's `x86_64-w64-mingw32.shared` toolchain. However, using `mxedeployqt` results in around a 1.5x larger distribution folder (~50mb more over 100mb using wine windeployqt). I'm trying to see if using the static mxe toolchain could help, but so far I'm hitting errors like https://public.kitware.com/pipermail/vtkusers/2010-September/062918.html - I'll see if I'm able to get it working.
